### PR TITLE
Add reference architecture pages

### PR DESCRIPTION
**Pull request by zchase**
Fixes: https://github.com/pulumi/pulumi-hugo/issues/643

This PR updates the solutions page and adds some reference architecture pages under the solutions URL.

### CODE REVIEW

File: themes/default/content/solutions/_index.md [hunk 1]
```diff
@@ -21,8 +21,33 @@ use_cases:
             - name: Lemonade
               link: /case-studies/lemonade
           description: |
-            Create reusable infrastructure components or self-service infrastructure platforms that empower your application developers to easily and quickly deploy
-            and use infrastructure they need to build applications.
+            Create reusable infrastructure components or self-service infrastructure platforms
+            that empower your application developers to easily and quickly deploy and use
+            infrastructure they need to build applications.
+
+            [Learn more](/solutions/shared-services-platforms/)
```
Comment (susanev) [inline themes/default/content/solutions/_index.md:28]:
```suggestion
            [Learn more about shared services platforms](/solutions/shared-services-platforms/)
```

File: themes/default/content/solutions/_index.md [hunk 2]
```diff
@@ -21,8 +21,33 @@ use_cases:
...
+        - name: Infrastructure CI/CD
+          case_studies:
+            - name: Atlassian
+              link: /case-studies/atlassian/
+            - name: Panther Labs
+              link: /case-studies/panther-labs/
+          description: |
+            Automate your infrastructure provisioning and management through a CI/CD pipeline. Run all infrastructure updates through a standard set of unit and integration tests, allowing reduced errors, greater security, and increased delivery velocity.
+
+            [Learn More](/solutions/infrastructure-ci-cd/)
```
Comment (susanev) [inline themes/default/content/solutions/_index.md:39]:
```suggestion
            [Learn more about CI/CD](/solutions/infrastructure-ci-cd/)
```

File: themes/default/content/solutions/_index.md [hunk 3]
```diff
@@ -21,8 +21,33 @@ use_cases:
...
+        - name: Terraform Migration
+          case_studies:
+            - name: Lemonade
+              link: /case-studies/lemonade/
+            - name: Panther Labs
+              link: /case-studies/sans-institute/
+          description: |
+            Modernize how your teams provision and manage infrastructure in order to adapt to the changes in technologies (containers, serverless), architectures (distributed), and software release automation associated with the cloud.
+
+            [Learn More](/solutions/terraform-migration)
```
Comment (susanev) [inline themes/default/content/solutions/_index.md:50]:
```suggestion
            [Learn more about Terraform migration](/solutions/terraform-migration)
```

**Review by susanev — APPROVED** [review]
a few things

- [ ] the how pulumi helps section on shared services looks really strange followed by the section above it; id consider moving this to the bottom or changing to 3 cards and doing a row
- [x] talk with a customer at bottom of shared services looks narrow, shouldn't it be wider? same issue on ci/cd page, and terraform page
- [x] @susanev missed two images, will add one more commit with their fixes

**Comment by zchase** [comment]
@susanev I filed a follow up issue for the "How Pulumi helps" section: https://github.com/pulumi/pulumi-hugo/issues/665

**Pull request merged**